core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -245,6 +259,27 @@ protected void doStop() throws Exception {
     return server;
   }
 
+  static boolean enableBasicAuth(String authMethod) {
+    return RestConfig.AUTHENTICATION_METHOD_BASIC.equals(authMethod);
+  }
+
+  static ConstraintSecurityHandler createSecurityHandler(String realm, List<String> roles) {
+    final ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
+    Constraint constraint = new Constraint();
+    constraint.setAuthenticate(true);
+    constraint.setRoles(roles.toArray(new String[roles.size()]));
```

Comment:
It turns out that passing `new String[0]` is more concise and faster (reference https://shipilev.net/blog/2016/arrays-wisdom-ancients/).